src/ui/Omnibox/OmniboxAnalytics.ts

```diff
@@ -0,0 +1,70 @@
+import { compact, filter, map, reduceRight, last } from 'underscore';
+import { IAnalyticsOmniboxSuggestionMeta } from '../Analytics/AnalyticsActionListMeta';
+
+export class OmniboxAnalytics {
+  public partialQueries: string[];
+  public suggestionRanking: number;
+  public suggestions: string[];
+  public partialQuery: string;
+
+  constructor() {
+    this.partialQueries = [];
+    this.suggestions = [];
+  }
+
+  public buildCustomDataForPartialQueries(index?: number, suggestions?: string[]): IAnalyticsOmniboxSuggestionMeta {
```

Comment:
The args are optional because, the Omnibox will set them, but when the `QuerySuggestPreview` will use it, they don't have this information so they wont use them.

Reply:
ok, I'll add `updateSuggestionsAndIndex` to the Omnibox and call it before we call this function.
this way, the args will no longer be necessary

Reply:
Ok great! In addition to using it before calling `buildCustomDateForPartialQueries`, another place where it would be useful is inside the `handleQuerySuggestGetFocus` method in Omnibox.ts.